examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,594 @@
+)
+
+scores = score_estimator(
+    glm_sev,
+    X_train[mask_train.values],
+    X_test[mask_test.values],
+    df_train[mask_train],
+    df_test[mask_test],
+    target="AvgClaimAmount",
+    weights="ClaimNb",
+)
+print(scores)
+
+##############################################################################
+#
+# Here, the scores for the test data call for caution as they are
+# significantly worse than for the training data indicating an overfit despite
+# the strong regularization.
+#
+# Note that the resulting model is the average claim amount per claim. As
+# such, it is conditional on having at least one claim, and cannot be used to
+# predict the average claim amount per policy in general.
+
+print("Mean AvgClaim Amount per policy:              %.2f "
+      % df_train["AvgClaimAmount"].mean())
+print("Mean AvgClaim Amount | NbClaim > 0:           %.2f"
+      % df_train["AvgClaimAmount"][df_train["AvgClaimAmount"] > 0].mean())
+print("Predicted Mean AvgClaim Amount | NbClaim > 0: %.2f"
+      % glm_sev.predict(X_train).mean())
+
+
+##############################################################################
+#
+# We can visually compare observed and predicted values, aggregated for
+# the drivers age (``DrivAge``).
+
+fig, ax = plt.subplots(ncols=1, nrows=2, figsize=(16, 6))
+
+plot_obs_pred(
+    df=df_train.loc[mask_train],
+    feature="DrivAge",
+    weight="Exposure",
+    observed="AvgClaimAmount",
+    predicted=glm_sev.predict(X_train[mask_train.values]),
+    y_label="Average Claim Severity",
+    title="train data",
+    ax=ax[0],
+)
+
+plot_obs_pred(
+    df=df_test.loc[mask_test],
+    feature="DrivAge",
+    weight="Exposure",
+    observed="AvgClaimAmount",
+    predicted=glm_sev.predict(X_test[mask_test.values]),
+    y_label="Average Claim Severity",
+    title="test data",
+    ax=ax[1],
+    fill_legend=True
+)
+plt.tight_layout()
+
+##############################################################################
+#
+# Overall, the drivers age (``DrivAge``) has a weak impact on the claim
+# severity, both in observed and predicted data.
+#
+# Pure Premium Modeling via a Product of Frequency and Severity
+# -------------------------------------------------------------
+# As mentioned in the introduction, the total claim amount per unit of
+# exposure can be modeled either as the product of the prediction of the
+# frequency model by the prediction of the severity model.
+#
+# To quantify the aggregate performance of this product model, one can compute
+# the mean deviance of the train and test data assuming a Compound
+# Poisson-Gamma distribution of the total claim amount. This is equivalent to
+# a Tweedie distribution with "power" parameter between 1 and 2.
+#
+# As we do not know the true value of the "power" parameter, we compute the
+# mean deviances for a grid of possible values of the "power" parameter,
+# hoping that a good model for one value of "power" will stay a good model for
+# another:
```

Comment:
I'm not sure I understand this paragraph. How should we interpret the results for different values of `power`, and how should we compare the results of the 2 approaches as done below?

Reply:
The topic was dicussed [in these comments](https://github.com/scikit-learn/scikit-learn/pull/14300#discussion_r333143202). Let's say our models a trained and therefore fixed. Which scoring function/metric should you then use to compare the models? You can use any tweedie deviance with `power` in [1,2) as those have the right support for y. There are 2 possibilities then:
1) Estimate `power` from the data, e.g. via maximum likelihood of the Tweedie distribution. Use this value for the metric.
2) Use a grid of `power` parameters for the metric, i.e. use a grid of differents metrics. Hopefully, those leed to similar conclusions as to which model performs better.

As the Tweedie likelihood is not implemented in python (and is highly non-trivial), approach 1) is not doable.

Reply:
Essentially, this is the result of the fact that we currently are not able to determine the optimal value of power by grid-search with the d2_metric since `power` impacts both the metric and the model. As mentioned below,

> I'm not sure I understand this paragraph. How should we interpret the results for different values of power, and how should we compare the results of the 2 approaches as done below?

So instead we evaluate the model on a grid of power values. Models are then compared identically as with multiple metrics (e.g. RMSE, MAE, mean_squared_log_error). One compares models metric by metric (or here power by power) with the caveat that some metrics (or powers) are more relevant than others. If one model is better for all metrics (here for all values of power) then it's probably better overall. 

Here it we put side by side test scores, we get,
```
subset                            Product model  TweedieRegressor
metric                                                                 
mean Tweedie deviance (p=1.5000)   8.640200e+01  8.618700e+01   
mean Tweedie deviance (p=1.7000)   3.920430e+01  3.917460e+01 
mean Tweedie deviance (p=1.8000)   3.148750e+01  3.148130e+01 
mean Tweedie deviance (p=1.9000)   3.420610e+01  3.420830e+01 
mean Tweedie deviance (p=1.9900)   1.996420e+02  1.996461e+02 
mean Tweedie deviance (p=1.9990)   1.892749e+03  1.892753e+03 
mean Tweedie deviance (p=1.9999)   1.882692e+04  1.882692e+04 
mean abs. error                    3.470205e+02  3.397084e+02 
mean squared error                 3.325902e+07  3.325471e+07
```
so overall that are pretty similar; MAE and deviance at (p=1.5) might be slightly better for TweedieRegressor.

Reply:
By the way, there is a whole theory about scoring functions/metrics. Let's try to be clear in this example but without overloading it. So maybe we should add @rth sentence: Compare the models metric by metric.

Reply:
Just as info - not for the example: If we were to compare all possible metrics (that are consistent) for the (conditional) expectation at once, we would end up using Murphy diagrams, see [paper](https://doi.org/10.1111/rssb.12154) and [R package](https://CRAN.R-project.org/package=murphydiagram). This may be something for the inspection module.